core/src/main/kotlin/endpoints/AgentManager.kt

```diff
@@ -549,10 +533,63 @@ class AgentManager(override val kodein: Kodein) : KodeinAware {
         }
     }
 
-    internal suspend fun AgentInfo.commitChanges() {
+    private suspend fun AgentInfo.commitChanges() {
         persistToDatabase()
+        notifyAgents(id)
+    }
+
+    internal suspend fun notifyAgents(agentId: String) {
         notifyAllAgents()
-        notifySingleAgent(id)
+        notifySingleAgent(agentId)
+    }
+
+    private suspend fun wrapInstanceBusy(
+        instanceKey: InstanceKey,
+        block: suspend AgentWsSession.() -> Unit,
+    ): Unit = getInstanceState(instanceKey)?.let { instanceState ->
+        val agentId = instanceKey.agentKey.agentId
+        updateInstanceStatus(instanceKey, AgentStatus.BUSY).also { instance ->
+            if (instance?.all { it.value.status == AgentStatus.BUSY } == true) {
```

Comment:
simplify boolean expression to if(instance.all { it.value.status == AgentStatus.BUSY })